qiskit/pulse/builder.py

```diff
@@ -0,0 +1,1986 @@
+
+        assert pulse_prog.ch_stop_time(d0) == pulse_prog.ch_start_time(d1)
+    """
+
+
+def _align(alignment: str = 'left') -> ContextManager[None]:
+    """General alignment context. Used by the :class:`_Builder` to choose the
+    default alignment policy.
+
+    Args:
+        alignment: Alignment scheduling policy to follow.
+            One of "left", "right" or "sequential".
+
+    Returns:
+        An alignment context that will schedule the instructions it contains
+        according to the selected alignment policy upon exiting the context.
+
+    Raises:
+        exceptions.PulseError: If an unsupported alignment context is selected.
+    """
+    if alignment == 'left':
+        return align_left()
+    elif alignment == 'right':
+        return align_right()
+    elif alignment == 'sequential':
+        return align_sequential()
+    else:
+        raise exceptions.PulseError('Alignment "{}" is not '
+                                    'supported.'.format(alignment))
+
+
+@_transform_context(transforms.group)
+def group() -> ContextManager[None]:
+    """Group the instructions within this context as a :class:`pulse.Schedule`.
+    fixing their relative time in parent contexts.
+
+    Examples:
+
+    .. jupyter-execute::
+
+        from qiskit import pulse
+
+        d0 = pulse.DriveChannel(0)
+        d1 = pulse.DriveChannel(1)
+        d2 = pulse.DriveChannel(2)
+
+        with pulse.build() as pulse_prog:
+            # will be ignored due to internal grouping
+            with pulse.align_sequential():
+                pulse.play(pulse.Constant(10, 1.0), d0)
+                with pulse.group():
+                    with pulse.align_left():
+                        # this pulse will start at t=10
+                        pulse.play(pulse.Constant(100, 1.0), d1)
+                        # this pulse will also start at t=10
+                        pulse.play(pulse.Constant(20, 1.0), d2)
+
+        assert pulse_prog.ch_start_time(d1) == pulse_prog.ch_start_time(d2)
+    """
+
+
+@contextmanager
+def inline() -> ContextManager[None]:
+    """Inline all instructions within this context into the parent context,
+    inheriting the scheduling policy of the parent context.
+
+    Examples:
+
+    .. jupyter-execute::
+
+        from qiskit import pulse
+
+        d0 = pulse.DriveChannel(0)
+        d1 = pulse.DriveChannel(1)
+        d2 = pulse.DriveChannel(2)
+
+        with pulse.build() as pulse_prog:
+            # will be ignored due to internal grouping
+            with pulse.align_left():
+                pulse.play(pulse.Constant(10, 1.0), d0)
+                with pulse.inline():
+                    with pulse.align_right():
+                        # this pulse will start at t=0
+                        pulse.play(pulse.Constant(100, 1.0), d1)
+                        # this pulse will also start at t=0
+                        pulse.play(pulse.Constant(20, 1.0), d2)
+
+        assert (pulse_prog.ch_start_time(d1) ==
```

Comment:
is this d0?

Reply:
Yes, thanks for catching. Fixed in 538a1000b